app/views/external_users/claims/case_details/_case_type_fields.html.haml

```diff
@@ -14,7 +14,7 @@
 
 - else
   - if display_elected_not_proceeded_signpost?(@claim)
-    - if Settings.main_hearing_date_enabled?
+    - if Settings.main_hearing_date_enabled_for_agfs? && @claim.agfs? || Settings.main_hearing_date_enabled_for_lgfs? && @claim.lgfs?
```

Comment:
As this is getting long, and repeated in three other places, it would be good to create a helper in `app/helpers/claims_helper.rb`.